### ISSUE TITLE
Add utility helpers for array parameters in MySQL resolvers

### ISSUE THREAD

**Issue by jamesw6811** [aws-amplify/amplify-cli#2028]
>**Note**: If your feature-request is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

**Is your feature request related to a problem? Please describe.**
It was a frustrating experience to sift through the documentation to find out how to do a batch query by ID on an Aurora DB in Amplify. It was especially confusing to figure out how to add a resolver to the stack - the documentation insinuates that this can be done by simply creating the resolver files, but in reality the stack file needs to be updated. The implementation was exacerbated significantly by not having any helper functions for dealing with MySQL parameters in the $util resolver library.

**Describe the solution you'd like**
I would like a batch query to be featured as an example in the documentation, I would like this to include editing the stack to add a custom resolver Aurora resource, and I would like to see a helper added to $util to convert an array input to a MySQL comma-separated values input. Ideally this would also handle sanitation.

**Describe alternatives you've considered**
The example could feature the code to do this instead of making it a helper. Hopefully you can find a more concise way to accomplish this than I did, pasted below in Additional Context.

**Additional context**
My custom resolver:

#if (!$util.isList($ctx.args.ids))
    $util.error("Not a list.")
#end

#foreach($num in $ctx.args.ids)
  #if (!$util.isNumber($num))
      $util.error("Not an integer input.")
  #end
#end

#set ($sqlargs = $ctx.args.ids[0].toString())
#set ($start = 1)
#set ($end = $ctx.args.ids.size())
#set ($end = $end - 1)
#set ($range = [$start..$end])
#foreach($i in $range)
  #set ($sqlargs = $sqlargs + "," + $ctx.args.ids[$i].toString())
#end

{
  "version": "2018-05-29",
  "statements":   ["SELECT * FROM Allergen WHERE id in ($sqlargs)"]
}

**Comment by attilah** [comment]
@jamesw6811, we sorry about the frustration you had along the way and we're grateful for the detailed feedback.

To add custom resources to your amplify project it is possible without modifying the generated stack: [here](https://aws-amplify.github.io/docs/cli-toolchain/graphql#api-category-project-structure). Just add your datasources to ```CustomResources.json``` file and they'll be created during deployment.

For custom resolvers [here](https://aws-amplify.github.io/docs/cli-toolchain/graphql#custom-resolvers) is the direct link.

The extension of the ```$util``` helper is a service side things, I'll ask the team and get back to you about this.

**Comment by jamesw6811** [comment]
Thanks @attilah ! I appreciate the response. I've poked around the repo, and you guys are doing an amazing job. Disclaimer - I am new to GraphQL and AppSync in general, so I'm learning a lot at once.

> To add custom resources to your amplify project it is possible without modifying the generated stack: here. Just add your datasources to CustomResources.json file and they'll be created during deployment.
![Capture](https://user-images.githubusercontent.com/2816551/62952895-8980d800-bdba-11e9-8dd6-0482aa512051.PNG)

The issue with this method is that I'm accessing an Aurora Serverless RDB - partially using the auto-generated resolvers/stack generated with `amplify api add-graphql-datasource`. My first attempt was to write my new custom resolvers in CustomResources.json, but the resolvers `dependOn` the database resource and access role. When I duplicated the database resource from the generated stack json over to CustomResources.json, `amplify push` breaks due to trying to duplicate the RDB resource on the backend. I was unable to find any documentation on how to import resources from the generated stack to add custom resolvers in the CustomResources stack. Do I have to choose between the auto-generated stack and a custom stack for any given resource? If so, that might be helpful to note in the docs.

More than anything, I think an example of more RDS custom queries/mutations, particularly one with an array-type parameter would be very helpful to newcomers.

It didn't help that the UI on the AWS AppSync Console was completely hiding the "attach resolver" button and resolver links due to its funky layout. I had to scroll right to reveal it. That's not your problem, but it may help to add it to the instructions for newcomers: https://github.com/aws-amplify/amplify-cli/issues/587
